### PR TITLE
Get job log by job_id in worker

### LINKED ISSUE #17886: Display running job logs in the jobservice dashboard
In jobservice dashboard,  add a link to display the current job service task's log, sometimes the log might be huge, displays the last 100 or 1000 rows of logs.

### PR DESCRIPTION
**Pull request by stonezdj**
Get the last 10MB of data if it exceeds 10MB
 Fixes #17886 
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by OrlinVasilev** [comment]
@stonezdj do we have issue or proposal that we can refer this PR to ?

File: src/controller/jobmonitor/monitor.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ type MonitorController interface {
 	PauseJobQueues(ctx context.Context, jobType string) error
 	// ResumeJobQueues resume the job queue by type
 	ResumeJobQueues(ctx context.Context, jobType string) error
+	JobLog(ctx context.Context, jobID string) ([]byte, error)
```
Comment (chlins) [inline src/controller/jobmonitor/monitor.go:73]: Follow the style of function name `Action + Object` as above, `GetJobLog` seems better.

File: src/server/v2.0/handler/jobservice.go [hunk 2]
```diff
@@ -196,3 +196,14 @@ func (j *jobServiceAPI) ActionPendingJobs(ctx context.Context, params jobservice
 	}
 	return jobservice.NewActionPendingJobsOK()
 }
+
+func (j *jobServiceAPI) GetJobLog(ctx context.Context, params jobservice.GetJobLogParams) middleware.Responder {
+	if err := j.RequireSystemAccess(ctx, rbac.ActionList, rbac.ResourceJobServiceMonitor); err != nil {
```
Comment (chlins) [inline src/server/v2.0/handler/jobservice.go:201]: `ActionGet`?

File: src/jobservice/logger/getter/file_getter.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ func (fg *FileGetter) Retrieve(logID string) ([]byte, error) {
 		return nil, errs.NoObjectFoundError(logID)
 	}
 
-	return os.ReadFile(fPath)
+	return tailLogFile(fPath, LogSizeLimit)
```
Comment (chlins) [inline src/jobservice/logger/getter/file_getter.go:37]: Add limitation here means no way to retrieve job's entire logs from outside(whatever API or inside manager/controller)?

File: src/controller/jobmonitor/monitor.go (hunk 1 above)
Comment (stonezdj) [reply src/controller/jobmonitor/monitor.go:73]: Updated

File: src/jobservice/logger/getter/file_getter.go (hunk 3 above)
Comment (stonezdj) [reply src/jobservice/logger/getter/file_getter.go]: Change the LogSizeLimit to be a configurable item in jobservice config file, if  0 or not configured, means no limitation on the log size returned.

**Review by chlins — APPROVED** [review]
lgtm

File: src/server/v2.0/handler/jobservice.go (hunk 2 above)
Comment (wy65701436) [reply src/server/v2.0/handler/jobservice.go:201]: +1

**Review by wy65701436 — APPROVED** [review]
lgtm

File: make/photon/prepare/templates/jobservice/config.yml.jinja [hunk 4]
```diff
@@ -51,3 +51,6 @@ metric:
   path: {{ metric.path }}
   port: {{ metric.port }}
 {% endif %}
+
+# the max size of job log returned by API, default is 10M
+max_retrieve_size_mb: 10
```
Comment (wy65701436) [inline make/photon/prepare/templates/jobservice/config.yml.jinja:56]: Would it be possible to use the value '-1' to represent an unlimited scenario?

Comment (stonezdj) [reply make/photon/prepare/templates/jobservice/config.yml.jinja:56]: <=0 will be considerd to unlimited, -1 is included.

**Pull request merged**